### ISSUE TITLE
Using aws_s3_bucket_object with kms_key_id causes "etag" issues

### ISSUE THREAD

**Issue by akarnani** [hashicorp/terraform#6698]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
0.6.16

### Affected Resource(s)
- aws_s3_bucket_object

### Terraform Configuration Files

```resource "aws_s3_bucket_object" "foo" {
    content = "abc"
    key = "foo"
    bucket = "${aws_s3_bucket.bar.id}"
    kms_key_id = "${aws_kms_key.main.arn}"
}```

### Expected Behavior
On successive applies in which the content changes, they should be applied correctly

### Actual Behavior

terraform appears to attempt to use an internally stored etag, however the check which prevents use of both `etag` and `kms_key_id` appears to fire and prevent the apply with the error message
```aws_s3_bucket_object.foo: Unable to specify 'kms_key_id' and 'etag' together because 'etag' wouldn't equal the MD5 digest of the raw object data```

### Steps to Reproduce
1. `terraform apply`
2. alter content
3. `terraform apply`

**Comment by coen-hyde** [comment]
Yep, I just experienced this

**Comment by coen-hyde** [comment]
This only happens on update. The first creation of the s3 object works fine. Looks like etag is being set in the statefile after creation regardless of kms. See: https://github.com/hashicorp/terraform/blob/76b3c74a5835659ce23abd66ce12bdf65c77f07a/builtin/providers/aws/resource_aws_s3_bucket_object.go#L168

**Comment by luhn** [comment]
I'm also having a problem with this.  Current workaround is to taint the resource.

**Comment by rvangraan** [comment]
Having the same problem. This only started happening when we upgraded to 0.7.2 from 0.6.x

**Comment by joseph-wortmann** [comment]
I am having the same problem after upgrading to 0.7.2

**Comment by catsby** [comment]
Hello friends – I just merged #9168, which adds an explicit conflict between `etag` and `kms_key_id`. You'll need to remove your `etag` if you're using `kms_key_id`. Sorry for the trouble!
